From 1b1a5b5a62ed7faf8b6c3d6749602ba7ad9df392 Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Tue, 24 Nov 2020 18:37:30 +0800
Subject: [PATCH] drm/rockchip: vop2: layer_sel and port_mux config done by
 port

If layer_sel or port_mux config done immediately, the layer
map and port mux may change when one frame is still in display
process.

Change-Id: I5611e75c56c89fd4fa02a069acc971c9266b05b6
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_vop2.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
index 06419b6f3b8b..f175eb6e4da5 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_vop2.c
@@ -3123,6 +3123,8 @@ static void vop2_setup_layer_mixer_for_vp(struct vop2_video_port *vp,
 	u8 layer_id, win_phys_id;
 	int i;
 
+	VOP_CTRL_SET(vop2, ovl_cfg_done_port, port_id);
+	VOP_CTRL_SET(vop2, ovl_port_mux_cfg_done_imd, 0);
 	for (i = 0; i < port_id; i++) {
 		used_layers += hweight32(vop2->vps[i].win_mask);
 		/*
-- 
2.35.3

